File: Perl/echon.pl
#Write a Perl script echon.pl which given exactly two arguments, an integer n and a string, prints the string n times. For example:
#./echon.pl 5 hello
#hello
#hello
#hello
#hello
#hello
#./echon.pl 0 nothing
#./echon.pl 1 goodbye
#goodbye

#!/usr/bin/perl

die "Usage: $0 <number of lines> <string>\n" if @ARGV != 2;
die "$0: argument 1 must be a non-negative integer\n" if $ARGV[0] < 0 or $ARGV[0] !~ /^\d+$/;

$n = 0;
$s = $ARGV[1];

while($n < $ARGV[0]){
        print "$s\n";
        $n++;
}

File: Python/echon.py
#!/usr/bin/python

import sys
import re

if(len(sys.argv) != 3):
    print('Usage: {} <number of lines> <string>'.format(sys.argv[0]))
    sys.exit(1)

match = re.match(r'^[0-9]+$', sys.argv[1])

if not match:
    print('{}: argument 1 must be a non-negative integer'.format(sys.argv[0]))
    sys.exit(1)

string = sys.argv[2]

for i in range(int(sys.argv[1])):
    print(string)

